rfcs/NNNN-decorator-roadmap.md

```diff
@@ -0,0 +1,217 @@
+
+##### Changes
+* Add new module with standard decorators
+    * Place them in a new module: `'lit/std-decorators.js'`
+* Re-export legacy decorators from `'lit/legacy-decorators.js'`
+* Add new module with a new implementation of _experimental_ decorators that work the same way standard decorators do:
+    * A _third_ set of decorators: `'lit/experimental-decorators.js'`
+    * They require `accessor`, replace the accessors on the prototype, do not call `static createProperty()`, store metadata in `[Symbol.metadata]`.
+    * This is so that users stuck on the experimental decorators setting (ie, google3) have a path forward when we remove the infrastructure that supports them.
+    * This decorator set will also be _use site_ compatible with standard decorators. Only the import will need to change to upgrade to standard decorators.
+    * There is no great way to emulate the standard `addInitializer()`, so we keep our `static addInitializer()` un-deprecated.
+* Non-core packages with decorators (@lit/labs) must follow a similar plan
+    * We can skip the legacy-decorators step with more aggressive breaking changes for labs packages.
+* Deprecate legacy experimental decorators, `static createProperty()`, `static getPropertyDescriptor()`, etc. but _not_ `static properties`.
+    * Since there are no native implementations of decorators yet, we don't quite yet want to deprecate the main developer-facing API for property declaration.
+* Deprecate the `noAccessor` property option.
+* Update `static elementProperties` to read from `[Symbol.metadata]`
+    * Standard decorators cannot call into the `static createProperty()` API, so they must place property options into `[Symbol.metadata]`. We should use that as the source-of-truth going forward.
+* Vend codemods to upgrade decorator usage:
+    * One to migrate to the new standard decorators
+    * One to migrate to the new experimental decorators
+
+#### II. Prefer standard decorators (breaking)
+
+This stage is still usable in non-decorator environments via `static properties`, but we move decorator modules around so that legacy use must use the non-default module names (`'lit/legacy-decorators.js'`) \
+
+##### Requirements
+
+* Stage I has landed.
+* Preferred: At least one browser has shipped decorators
+
+##### Changes
+
+* Move standard decorators implementation to <code>'lit/decorators.js'</code>
+* Re-export standard decorators from <code>'lit/std-decorators.js'</code>
+* Deprecate <code>static properties</code>
+* Deprecate <code>static addInitializer()</code>
+* Vend codemod to migrate standard decorator imports
+
+#### III. Remove legacy decorators (breaking)
+
+This stage requires decorators for creating properties. It is no longer usable in non-decorator-supporting environments without transpilation.
+
+##### Requirements
+
+* Native decorators are shipping in all major browsers.
+* Stage II has landed
+
+##### Changes
+
+* Remove previously deprecated APIs
```

Comment:
I assume this includes `'lit/legacy-decorators.js';`?

Reply:
yes :)